### PR TITLE
D17 66/frontend

### PR DESCRIPTION
**Pull request by irosikoni**
[Screencast from 2024-05-19 18-22-59.webm](https://github.com/Spolka-Zlo/D17Map-Frontend/assets/108950456/8f61b51a-614f-4f36-b5cb-8505f3e05dad)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,8 +9,12 @@
     "lint": "next lint"
   },
   "dependencies": {
+    "date-fns": "^3.6.0",
+    "import": "^0.0.6",
```
Comment (wleklinskimateusz) [inline package.json]: wut?

File: package.json [hunk 2]
```diff
@@ -9,8 +9,12 @@
     "lint": "next lint"
   },
   "dependencies": {
+    "date-fns": "^3.6.0",
```
Comment (wleklinskimateusz) [inline package.json]: is this used anywhere?

File: src/app/calendar/_components/CalendarSection.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+"use client";
+import { Calendar } from "@/components/Calendar";
+import { useState } from "react";
+import { CalendarTimeManager } from "./CalendarTimeManager";
+import { CalendarDaySection } from "./CalendarDaySection";
+
+enum ReservationType {
```
Comment (wleklinskimateusz) [inline src/app/calendar/_components/CalendarSection.tsx]: Enums are evil. Don't use them in typescript
https://www.youtube.com/watch?v=jjMbPt_H3RQ

File: src/app/calendar/_components/CalendarSection.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  Lecture = "Lecture",
+  Consultation = "Consultation",
+  Exam = "Exam",
+}
+
+type Reservation = {
```
Comment (wleklinskimateusz) [inline src/app/calendar/_components/CalendarSection.tsx:9]: You don't use it anywhere here

File: src/app/calendar/_components/CalendarSection.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <section className="flex">
+      <div className="grid grid-cols-2 maxSM:grid-cols-1 maxSM:grid-rows-2 rounded-lg">
+        <Calendar
+          mode="single"
+          selected={selectedDate}
+          onDayClick={(day: Date) => {
+            console.log("compare", day.getDay === selectedDate?.getDay);
+            setIsOpen((prev) =>
```
Comment (wleklinskimateusz) [inline src/app/calendar/_components/CalendarSection.tsx]: why make it a callback if you don't use prev anymore

File: src/app/calendar/_components/CalendarSection.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    new Date()
+  );
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <section className="flex">
+      <div className="grid grid-cols-2 maxSM:grid-cols-1 maxSM:grid-rows-2 rounded-lg">
+        <Calendar
+          mode="single"
+          selected={selectedDate}
+          onDayClick={(day: Date) => {
```
Comment (wleklinskimateusz) [inline src/app/calendar/_components/CalendarSection.tsx]: you shouldn't need to type it implicitly as Date
```suggestion
          onDayClick={(day) => {
```

File: src/app/calendar/_components/CalendarSection.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+            console.log("compare", day.getDay === selectedDate?.getDay);
```
Comment (wleklinskimateusz) [inline src/app/calendar/_components/CalendarSection.tsx]: I don't believe you need this console.log here

File: src/app/calendar/_components/CalendarTimeManager.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+type CalendarTimeManagerProps = {
+  isOpen: boolean;
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+  className?: string;
+};
+
+export function CalendarTimeManager({
+  isOpen,
+  setIsOpen,
+  className,
```
Comment (wleklinskimateusz) [inline src/app/calendar/_components/CalendarTimeManager.tsx]: you have destructure className, but not use it anywhere

File: src/app/calendar/_components/CalendarTimeManager.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+          <OrangeButton
+            text="Sprawdź dostępność"
+            className="bg-white hover:bg-secondary text-secondary hover:text-primary hover:shadow-primary"
+            textClassName=""
+          />
+        </div>
+      )}
+      {isOpen && (
+        <div className="absolute p-4 top-5 right-2">
+          <XButton className="z-10" onClickHandler={() => setIsOpen(false)} />
```
Comment (wleklinskimateusz) [inline src/app/calendar/_components/CalendarTimeManager.tsx]: I would change the name of this component to CloseButton

File: src/app/calendar/_components/CalendarTimeManager.tsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    <div
+      className={twMerge(
+        "z-0 sm:rounded-r-lg maxSM:rounded-b-lg bg-secondary relative",
+        !isOpen &&
+          "sm:w-0 maxSM:h-0 duration-700 transition-[width] maxSM:transition-[height]",
+        isOpen &&
+          "sm:w-full maxSM:h-full sm:-ml-2 maxSM:-mt-2 transition-[width] maxSM:transition-[height] duration-700"
+      )}
+    >
+      {isOpen && (
```
Comment (wleklinskimateusz) [inline src/app/calendar/_components/CalendarTimeManager.tsx]: You have duplicate `isOpen && ()` either merge it into one statement, or move it all to a seperate component with an early return (`if (!isOpen) return null`)

File: src/components/Calendar.tsx [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+          "bg-secondary rounded-full transition-colors duration-300 text-primary hover:bg-secondary/50",
+        day_today: "border-2 border-secondary rounded-full",
+        day_outside: "opacity-50 aria-selected:opacity-30",
+        day_disabled: " text-mapGrey/70",
+        day_range_middle: "",
+        day_hidden: "invisible",
+        ...classNames,
+      }}
+      components={{
+        IconLeft: ({ ...props }) => <ChevronLeft className="h-4 w-4" />,
```
Comment (wleklinskimateusz) [inline src/components/Calendar.tsx]: you're not using `{...props}` anywhere
```suggestion
        IconLeft: () => <ChevronLeft className="h-4 w-4" />,
```

File: src/components/Calendar.tsx [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+        IconRight: ({ ...props }) => <ChevronRight className="h-4 w-4" />,
```
Comment (wleklinskimateusz) [inline src/components/Calendar.tsx]: the same thing

File: src/components/Calendar.tsx [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+      }}
+      {...props}
+    />
+  );
+}
+Calendar.displayName = "Calendar";
```
Comment (wleklinskimateusz) [inline src/components/Calendar.tsx]: it shouldn't be necesery in this component
```suggestion
```

File: src/components/Calendar.tsx [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+
+export { Calendar };
```
Comment (wleklinskimateusz) [inline src/components/Calendar.tsx]: avoid exporting things in this way. Use "export" keyword next to "function"

**Review by wleklinskimateusz — CHANGES_REQUESTED** [review]
I think you may have bun.lock not up to date because when I've installed dependencies it got overridden

File: src/app/calendar/_components/CalendarSection.tsx (hunk 4 above)
Comment (irosikoni) [reply src/app/calendar/_components/CalendarSection.tsx:9]: its prepared for later extension

**Review by wleklinskimateusz — APPROVED** [review]
LGTM

**Review by yoinker57 — APPROVED** [review]

**Pull request merged**